test/watch.js

```diff
@@ -44,6 +44,24 @@ describe('gulp', function() {
       updateTempFile(tempFile);
     });
 
+    it('should not call the function when no file changes: no options', function(done) {
+      var tempFile = path.join(outpath, 'watch-func.txt');
+      var called = false;
+
+      createTempFile(tempFile);
+
+      var watcher = gulp.watch(tempFile, function(cb) {
+        watcher.end();
+        called = true;
```

Comment:
Throw an error or call `should.fail()` here instead, "expected true to equal false" isn't a descriptive error